AnonUpload/src/main/java/com/theironyard/controllers/AnonFileController.java

```diff
@@ -24,18 +26,35 @@
     @Autowired
     AnonFileRepository repository;
 
+    public static final int LIMIT = 5;
+
     @RequestMapping(path = "/upload", method = RequestMethod.POST)
-    public void upload(MultipartFile file, HttpServletResponse response) throws IOException {
+    public void upload(MultipartFile file, HttpServletResponse response, String permanent, String comment,String password) throws IOException, PasswordStorage.CannotPerformOperationException {
         File dir = new File("public/files");
         if(!dir.exists()) {
             dir.mkdirs();
         }
-
         File f = File.createTempFile("anonfile", file.getOriginalFilename(), dir);
         FileOutputStream fos = new FileOutputStream(f);
         fos.write(file.getBytes());
-
         AnonFile anonFile = new AnonFile(f.getName(), file.getOriginalFilename());
+        if(anonFile.getPassword() == null){
+            anonFile.setPassword(PasswordStorage.createHash(password));
+        }
+        if(!comment.isEmpty()){
+            anonFile.setComment(comment);
+        }
+        if(permanent == null){
+            anonFile.setPermanent(false);
+        }else{
+            anonFile.setPermanent(true);
+        }
+
+        long nonPermFiles = repository.countByPermanentFalse();
+        if(nonPermFiles >= LIMIT ){
```

Comment:
What if the files are way above the limit?  This will only ever delete 1 which will never catch up.